website/test/functional/test_views.py

```diff
@@ -0,0 +1,58 @@
+from lxml import html
+from django.urls import reverse
+
+
+def test_login(db, client, data):
+    response = client.post(
+        '/login/', {'username': 'test', 'password': 'test'}
+    )
+    assert response.status_code == 302
+
+
+def test_login_fail(db, client, data):
+    response = client.post(
+        '/login/', {'username': 'test', 'password': 'wrongpassword'}
+    )
+    assert response.status_code == 200
+    assert b'Please enter a correct username and password.' in response.content
```

Comment:
Compare str vs response.text

Reply:
response.text not identified, so instead of that decoding is used:
`assert 'Please enter a correct username and password.' in response.content.decode()`